app/controllers/comments_controller.rb

```diff
@@ -1,25 +1,24 @@
 class CommentsController < ApplicationController
   include HackernewsClone::VoteHelper
-  include HackernewsClone::CommentHelper
 
-  before_action :authenticate_to_submit, only: :new
   before_action :authenticate_user!, only: :create
+  before_action :authenticate_to_submit, only: :new
 
-  helper_method :comments_parent, :comment
+  helper_method :new_comment, :commentable, :comment
 
   def new
   end
 
   def create
-    @comment = current_user.comments.build(comment_params)
-    parent = @comment.commentable
-    if @comment.save
-      redirect_to parent
+    @new_comment = current_user.comments.build(comment_params)
+    commentable = @new_comment.commentable
```

Comment:
I'd use the `new_comment` reader method here and below since the ivar was set above.